### PR TITLE
feat: tools ai

### PR DESCRIPTION
**Pull request by 0xDiscotech**
## Summary by CodeRabbit

* **New Features**
  * Introduced custom error messages for invalid refund addresses and unrecognized tokens in the bridge interface.
  * Added configuration and documentation files outlining coding, testing, security, and performance guidelines for TypeScript, Solidity, and React projects.

* **Bug Fixes**
  * Improved validation in the token bridge to ensure refund addresses are not zero and tokens are properly recognized.

* **Tests**
  * Added and enhanced test cases to cover invalid refund address handling and stricter token validation, increasing test coverage for edge cases.

* **Documentation**
  * Updated and expanded project documentation and guidelines for code structure, error handling, testing, security, and best practices.
  * Removed outdated risk information regarding re-entrancy from the README.

### CODE REVIEW

**Comment by 0xDiscotech** [comment]
@coderabbit full review

File: .cursor/rules/Solidity/solidity-security.mdc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+## Security Patterns
+- Use CEI pattern (Checks-Effects-Interactions)
+- Prefer custom errors over require statements
+- Include comprehensive NatSpec documentation
+- Emit events for storage changes
+- Inherit interfaces last to avoid linearization issues
+
+## Best Practices
+- Use SafeMath or unchecked blocks appropriately
+- Validate all inputs
```
[the remark this answers is not in the document]
Comment (0xDiscotech) [reply .cursor/rules/Solidity/solidity-security.mdc]: [bbfc7ed](https://github.com/defi-wonderland/op-ref-tokens/pull/23/commits/bbfc7ed450e9a9a9d73c8a9ca3e0c00cd01bd495)

File: .cursor/rules/solidity-best-practices.mdc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+description: 
+globs: 
+alwaysApply: false
+---
+---
+description: Enforces best practices for writing Solidity code, including naming conventions, security patterns, and gas optimization techniques.
+globs:
+  - "**/*.sol"
+alwaysApply: false
+---
```
[the remark this answers is not in the document]
Comment (0xDiscotech) [reply .cursor/rules/solidity-best-practices.mdc]: [79db273](https://github.com/defi-wonderland/op-ref-tokens/pull/23/commits/79db2736bcc4e6563a5d3085a38b51023d356738)

File: .cursor/rules/general-rules.mdc [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+description: 
+globs: 
+alwaysApply: false
+---
```
[the remark this answers is not in the document]
Comment (0xDiscotech) [reply .cursor/rules/general-rules.mdc]: [79db273](https://github.com/defi-wonderland/op-ref-tokens/pull/23/commits/79db2736bcc4e6563a5d3085a38b51023d356738)

File: .cursor/rules/testing.mdc [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+globs: 
+alwaysApply: false
+---
+---
+description: Guidelines for testing and quality assurance, including test strategies, tooling, and coverage goals.
+globs:
+  - "**/*.t.sol"
+  - "test/**/*"
+alwaysApply: false
+---
```
[the remark this answers is not in the document]
Comment (0xDiscotech) [reply .cursor/rules/testing.mdc]: [79db273](https://github.com/defi-wonderland/op-ref-tokens/pull/23/commits/79db2736bcc4e6563a5d3085a38b51023d356738)

[the remark this answers is not in the document]
Comment (0xDiscotech) [reply .cursor/rules/testing.mdc]: [c8c6832](https://github.com/defi-wonderland/op-ref-tokens/pull/23/commits/c8c68327d20b80fdc99a373f8fc09ad4a3ee8f11)

[the remark this answers is not in the document]
Comment (0xDiscotech) [reply .cursor/rules/testing.mdc]: and [746dcb4](https://github.com/defi-wonderland/op-ref-tokens/pull/23/commits/746dcb4b4b9204a8b992babf15a0c69907d8a59e)

**Review by ashitakah — APPROVED** [review]
👍
